CybridSDK/Components/TransferComponent/View/TransferViewModel.swift

```diff
@@ -0,0 +1,150 @@
+//
+//  Created by Erick Sanchez Perez on 23/12/22.
+//
+
+import Foundation
+import CybridApiBankSwift
+
+class TransferViewModel: NSObject {
+
+    // MARK: Private properties
+    private var dataProvider: AccountsRepoProvider & ExternalBankAccountProvider & QuotesRepoProvider & TradesRepoProvider & AssetsRepoProvider
+    private var logger: CybridLogger?
+
+    // MARK: Internal properties
+    internal var customerGuid = Cybrid.customerGUID
+    internal var assets: [AssetBankModel] = []
+    internal var accounts: Observable<[AccountBankModel]> = .init([])
+    internal var externalBankAccounts: Observable<[ExternalBankAccountBankModel]> = .init([])
+    internal var fiatBalance: Observable<String> = .init("")
+    internal var currentQuote: Observable<QuoteBankModel?> = .init(nil)
+    internal var currentTrade: Observable<TradeBankModel?> = .init(nil)
+
+    // MARK: Public properties
+    var uiState: Observable<TransferViewController.ViewState> = .init(.LOADING)
+    var currentFiatCurrency = "USD"
+
+    // MARK: Constructor
+    init(dataProvider: AccountsRepoProvider & ExternalBankAccountProvider & QuotesRepoProvider & TradesRepoProvider & AssetsRepoProvider,
+         logger: CybridLogger?) {
+
+        self.dataProvider = dataProvider
+        self.logger = logger
+    }
+
+    // MARK: ViewModel Methods
+    func getAccounts() {
+        self.fetchAssets()
+    }
+
+    func fetchAssets() {
+
+        self.dataProvider.fetchAssetsList { [weak self] assetsResponse in
+
+            switch assetsResponse {
+
+            case .success(let assets):
+
+                self?.logger?.log(.component(.accounts(.pricesDataFetching)))
+                self?.assets = assets
+                self?.fetchAccounts()
+
+            case .failure:
+                self?.logger?.log(.component(.accounts(.pricesDataError)))
+            }
+        }
+    }
+
+    func fetchAccounts() {
+
+        self.dataProvider.fetchAccounts(customerGuid: self.customerGuid) { [weak self] accountsResponse in
+
+            switch accountsResponse {
+            case .success(let accountList):
+                self?.logger?.log(.component(.accounts(.accountsDataFetching)))
+                self?.accounts.value = accountList.objects
+                self?.fetchExternalAccounts()
+
+            case .failure:
+                self?.logger?.log(.component(.accounts(.accountsDataError)))
+            }
+        }
+    }
+
+    func fetchExternalAccounts() {
+
+        self.dataProvider.fetchExternalBankAccounts { [weak self] accountsResponse in
+            switch accountsResponse {
+            case .success(let accountList):
+                self?.logger?.log(.component(.accounts(.accountsDataFetching)))
+                self?.externalBankAccounts.value = accountList.objects
+                self?.uiState.value = .ACCOUNTS
+
+            case .failure:
+                self?.logger?.log(.component(.accounts(.accountsDataError)))
+            }
+        }
+    }
```

Comment:
Are we adding this same code in multiple models? If so, we should create common services that do this for us and reuse them everyone.

Reply:
Sure!!